platform/util/platform-util-unavailableurls/src/test/java/org/codice/ddf/platform/util/http/UnavailableUrlsTest.java

```diff
@@ -0,0 +1,185 @@
+            }
+        };
+    }
+
+    @Test
+    public void testRandomContains() {
+        UnavailableUrls set = initSet(200);
+        assertThat(set.contains("random"), is(false));
+    }
+
+    @Test
+    public void testInitialContains() {
+        UnavailableUrls set = initSet(200);
+        set.add("hello");
+        assertThat(set.contains("hello"), is(true));
+    }
+
+    private class SchedulerCapture {
+        public final Runnable runnable;
+
+        public final long timeout;
+
+        public final TimeUnit unit;
+
+        public SchedulerCapture(Runnable r, long to, TimeUnit tu) {
+            this.runnable = r;
+            this.timeout = to;
+            this.unit = tu;
+        }
+    }
+
+    // capture the call to the scheduler and return the parameters
+    private SchedulerCapture capture() {
+
+        ArgumentCaptor<Runnable> runnable = ArgumentCaptor.forClass(Runnable.class);
+        ArgumentCaptor<Long> timeout = ArgumentCaptor.forClass(Long.class);
+        ArgumentCaptor<TimeUnit> unit = ArgumentCaptor.forClass(TimeUnit.class);
+
+        Mockito.verify(scheduler).schedule(runnable.capture(), timeout.capture(), unit.capture());
+        Mockito.reset(scheduler);
+
+        return new SchedulerCapture(runnable.getValue(), timeout.getValue(), unit.getValue());
+    }
+
+    // test the contents of the set after a ping
+    // statusCode - the http status code the ping should return
+    // present - is the url suppose to be present?
+    private void testPing(int statusCode, boolean present) {
+
+        UnavailableUrls set = initSet(statusCode);
+
+        set.add("hello");
+
+        capture().runnable.run(); // force URL ping
+
+        assertThat(set.contains("hello"), is(present));
+    }
+
+    @Test
+    // if the ping returns ok, the url shouldn't be present
+    public void testSuccessfulPing() {
+        testPing(HttpURLConnection.HTTP_OK, false);
+    }
+
+    @Test
+    // if the ping returns not ok, the url should still be present
+    public void testUnsuccessfulPing() {
+        testPing(HttpURLConnection.HTTP_NOT_FOUND, true);
+    }
+
+    /**
+     * Check if a list of numbers grows exponentially or greater upto a max backoff.
+     *
+     * @param list - list of two or more positive integers
+     * @return true if sequence is exponential, false otherwise
+     */
+    private boolean isExponentialBackoff(long[] list) {
+        if (list.length < 2) {
+            return false;
+        }
+
+        double base = list[1] / (double) list[0];
+        long max = list[list.length - 1];
+
+        if (!(base > 1)) {
+            return false;
+        }
+
+        for (int i = 1; i < list.length - 1; i++) {
+            double changeOfBase = Math.log(list[i]) / Math.log(base);
+            if (changeOfBase < i && list[i] != max) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    @Test
+    public void testIsExponentialBackoff() {
```

Comment:
Is this a test for a test method? Doesn't hurt anything but probably not necessary.